### ISSUE TITLE
[Feature Request] Internationalized domain name (IDN) support

### ISSUE THREAD

**Issue by PeterDaveHello** [ypcrts/fqdn#25]
Hello,

I just found this useful python tool, it looks great in most cases! Not sure is [IDN](https://en.wikipedia.org/wiki/Internationalized_domain_nam) support will be something you'd also like to have, it'll be awesome if it's also supported.

Currently, IDN will be considered as an invalid FQDN:
```
$ python
Python 3.6.9 (default, Jul 17 2020, 12:50:27) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from fqdn import FQDN
>>> FQDN('Bücher.example').is_valid
False
```

Thank you!

**Comment by ypcrts** [comment]
Yes I do want IDN support, but without breaking old use cases. I’ll see what Santa is bringing this year.

**Comment by PeterDaveHello** [comment]
Thank you @ypcrts 😃

**Comment by ypcrts** [comment]
@PeterDaveHello 

What is your use case for this?  You want `FQDN('Bücher.example').is_valid` to return `True`. What do you want for the internal state? For example, do you want `FQDN('Bücher.example').relative` to return `xn--bcher-kva.example` or `Bücher.example`? Why?

Really, what I'm asking is, how is your use case different from #12 ?

**Comment by PeterDaveHello** [comment]
To me personally I prefer `.relative()` to return `xn--bcher-kva.example`, because it's more comparable, one of the issues of IDN is the identifyability by human eyes, but I may need to dig more about how IDN is saved in the most popular DNS service providers.

**Comment by PeterDaveHello** [comment]
I'm not pretty sure but this may just be duplicated with #12

**Comment by PeterDaveHello** [comment]
cc @ross & https://github.com/octodns/octodns/issues/199, octodns may be able to support IDN via the implementation here(and may also contribute back), I've already use this fqdn package in octodns to help fdqn validation (https://github.com/octodns/octodns/pull/631)!
